include/zephyr/sys/internal/time.h

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2024 Bjarki Arge Andreasen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_SYS_INTERNAL_TIME_H_
+#define ZEPHYR_INCLUDE_SYS_INTERNAL_TIME_H_
+
+#include <zephyr/types.h>
+#include <zephyr/drivers/rtc.h>
+
+/** Validate timestamp can be converted to datetime */
+bool sys_time_validate_timestamp(const int64_t *timestamp);
```

Comment:
Please call this `sys_rtc_time...` or any other `_rtc_` prefix or infix, as this API is closely  coupled to rtc.h. The time namespace is so crowded already in Zephyr that generic namespaces like sys_time should not be occupied by specific libraries.